tdp/core/variables/service_variables.py

```diff
@@ -242,7 +236,7 @@ def validate_schema(self, variables: Variables, schema: dict) -> None:
         except exceptions.ValidationError as e:
             raise InvalidSchema("Schema is invalid", variables.name) from e
 
-    def validate(self) -> None:
+    def validate(self, schema: dict) -> None:
```

Comment:
I don't understand this PR. Now when I have an instance of `ServiceVariables` I need to give a `dict` representing the schema every time I want to validate the variables? But the schema doesn't change between each validation.

Reply:
With #320 in mind (preventing default variables copy), we'll need to have access to default variables in the schema validation logic. Both default variables and schema come from the `Collections` object, hence, it seemed better for me to pass a `collections` object as an argument instead of storing everything in the `ServiceVariables` object.

I did this PR to prepare this change.